### PR TITLE
TF Plugin Docs - migrating to new structure

### PR DESCRIPTION
**Pull request by sheneska**
### Description

This PR covers the following:

- migrates files from the legacy structure `(docs/r)` to the `tfplugindocs` supported `templates/` structure
- generates templated files for each resource & data source as well as example.tf files 

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
...
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: templates/resources/secret_v1.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+The resource provides mechanisms to inject containers with sensitive information, such as passwords, while keeping containers agnostic of Kubernetes. Secrets can be used to store sensitive information either as individual properties or coarse-grained entries like entire files or JSON blobs. The resource will by default create a secret which is available to any pod in the specified (or default) namespace.
+
+~> Read more about security properties and risks involved with using Kubernetes secrets: [Kubernetes reference](https://kubernetes.io/docs/concepts/configuration/secret/#information-security-for-secrets)
+
+~> **Note:** All arguments including the secret data will be stored in the raw state as plain-text. [Read more about sensitive data in state](/docs/state/sensitive-data.html).
+
+## Example Usage
+
+{{tffile "/Users/sheneskawilliams/dev/terraform-provider-kubernetes/examples/resources/secret_v1/example_1.tf"}}
```
Comment (arybolovlev) [inline templates/resources/secret_v1.md.tmpl:22]: Please replace absolute paths with relative ones.

File: templates/resources/secret_v1.md.tmpl [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+- `create` - Default `1 minute`
+
+## Import
+
+Secret can be imported using its namespace and name, e.g.
+
+```
+$ terraform import kubernetes_secret_v1.example default/my-secret
+```
+---
```
Comment (arybolovlev) [inline templates/resources/secret_v1.md.tmpl:92]: It looks like the content of each file duplicates. Starting from this line and till the end of the file is a duplicate.

File: templates/resources/namespace_v1.md.tmpl [hunk 3]
```diff
@@ -0,0 +1,127 @@
+---
+subcategory: "core/v1"
+page_title: "Kubernetes: kubernetes_namespace_v1"
```
Comment (arybolovlev) [inline templates/resources/namespace_v1.md.tmpl:3]: Can we replace `kubernetes_namespace_v1` with `{{ .Name }}`?

File: templates/resources/namespace_v1.md.tmpl [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+description: |-
+  Kubernetes supports multiple virtual clusters backed by the same physical cluster. These virtual clusters are called namespaces.
```
Comment (arybolovlev) [inline templates/resources/namespace_v1.md.tmpl:5]: Can we replace description here with `{{ .Description }}`?

File: kubernetes/resource_kubernetes_namespace_v1.go [hunk 5]
```diff
@@ -39,7 +40,7 @@ func resourceKubernetesNamespaceV1() *schema.Resource {
 			},
 		},
 		Timeouts: &schema.ResourceTimeout{
-			Delete: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(2 * time.Minute),
```
Comment (arybolovlev) [inline kubernetes/resource_kubernetes_namespace_v1.go:43]: I think you used this one to validate doc-gen, could you please revert this change or share more details? :)

File: templates/resources/namespace_v1.md.tmpl [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+{{ .Description }}
+
+{{ .SchemaMarkdown	}}
+
+## Example Usage
+
+{{tffile "examples/resources/namespace_v1/example_1.tf"}}
+
+## Argument Reference
```
Comment (arybolovlev) [inline templates/resources/namespace_v1.md.tmpl:17]: This part overlaps with what is rendered by `{{ .SchemaMarkdown }}`. Shell we remove it from here?

**Review by arybolovlev — CHANGES_REQUESTED** [review]
I have left a few comments, could you please take a look? Most of them apply to all files in this PR, so I picked up only one and commented there.

One more thing that I have noticed -- doc-gen does not pick up default values. Do you know if there are plans to fix that? It might be crucial for some resources and timeouts.

All in all, great efforts, thank you!

**Review by SarahFrench — COMMENTED** [review]
👋 Hi Sheneska, I saw your post on Slack and took a look at the problem.

It looks like you need to update these files to reflect the new locations of the markdown files:
- This line in [.github/workflows/markdown-link-check.yaml](https://github.com/hashicorp/terraform-provider-kubernetes/blob/4f75983bd1e39f1444337934336673ac26d11a31/.github/workflows/markdown-link-check.yaml#L11)
- Times that [the Makefile](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/GNUmakefile) references the old `./website` path. The Makefile is used in the [checkers-and-linters.yaml file](https://github.com/hashicorp/terraform-provider-kubernetes/blob/4f75983bd1e39f1444337934336673ac26d11a31/.github/workflows/checkers-and-linters.yml#L31-L40), which is linked to one of the checks that's failing.

**Review by arybolovlev — APPROVED** [review]

**Pull request merged**